Return the ignored error

EugeneChung opened the pull request:
The error from c.InitSecContext() is ignored.

Commit:
Return the ignored error

The error from c.InitSecContext is ignored.

beltran commented on gssapi.go:
```diff
@@ -242,7 +242,7 @@ func initClientContext(c *GSSAPIContext, service string, inputToken []byte) erro
 	c.contextId = contextId
 	c.availFlags = outputRetFlags
-	return nil
+	return err
```
Instead of returning err here can you add 
```
if err != nil {
    return err
}
```
before `c.token`

beltran reviewed (changes requested):
This seems to make to make the test TestGSSAPIMechanism in sasl_gssapi_test.go fail, can you just remove the lines that are making this fail from there?
Those would be:
```
for _, input := range [][]byte{[]byte("Ahjdskahdjkaw12kadlsj"), []byte("0"), nil} {
        client.Step(input)
}
```

Commit:
return error before setting context

EugeneChung commented:
> This seems to make to make the test TestGSSAPIMechanism in sasl_gssapi_test.go fail, can you just remove the lines that are making this fail from there?
> Those would be:
> 
> ```
> for _, input := range [][]byte{[]byte("Ahjdskahdjkaw12kadlsj"), []byte("0"), nil} {
>         client.Step(input)
> }
> ```

```
	} else if m.negotiationStage == 1 {
		err := initClientContext(m.context, m.service+"/"+m.host, challenge)
		if err != nil {
			log.Fatal(err)
			return nil, err
		}
```
of gssapi.go makes it fail because of the error from initClientContext(). I get it.

Commit:
remove calling client.Step()

After making it return the error from initClientContext() of gssapi.go properly, calling client.Step() seems always fail here.

beltran commented:
Thank you for your contribution!
